1-JavaScript/Week2/assignment/ex4-shoppingCart.js

```diff
@@ -1,33 +1,21 @@
 'use strict';
-/*------------------------------------------------------------------------------
-Full description at: https://github.com/HackYourFuture/Assignments/tree/main/1-JavaScript/Week3#exercise-4-shopping-at-the-supermarket
 
-Let's do some grocery shopping! We're going to get some things to cook dinner
-with. However, you like to spend money and always buy too many things. So when 
-you have more than 3 items in your shopping cart the first item gets taken out.
-
-1. Create an array called `shoppingCart` that holds the following strings: 
-   "bananas" and "milk".
-
-2. Complete the function named `addToShoppingCart` as follows:
-
-   - It should take one argument: a grocery item (string)
-   - It should add the grocery item to `shoppingCart`. If the number of items is
-     more than three remove the first one in the array.
-   - It should return a string "You bought <list-of-items>!", where 
-     <list-of-items>is a comma-separated list of items from the shopping cart 
-     array.
-
-3. Confirm that your code passes the unit tests.
------------------------------------------------------------------------------*/
 const shoppingCart = ['bananas', 'milk'];
 
 // ! Function to be tested
-function addToShoppingCart(/* parameters go here */) {
-  // TODO complete this function
+function addToShoppingCart(item) {
+  
+  if (item) {
+    shoppingCart.push(item);
+    if (shoppingCart.length > 3) {
+      shoppingCart.shift();
+    }
+  }
+  return `You bought ${shoppingCart.join(', ')}!` ;
```

Comment:
Here's one more of those oddly placed spaces that I would personally avoid.

```suggestion
  return `You bought ${shoppingCart.join(', ')}!`;
```

Reply:
in this exercise when i run "npm run test" i get an error (a unit test file was not provided)!

Reply:
@Nimasaghi-dev sorry, I meant to say `npm run it` and not `test`.

Reply:
i just complete all you have mentioned, and run the "npm run it".